### PR TITLE
ObjectInspector improvements (FE-1230)

### PR DESCRIPTION
**Pull request by hbenl**
- add support for inspecting the internal state of `Promise` and `Proxy` objects
- fix the display of `Set` contents
- add checks to the object_preview-01 e2e test for displaying the contents of `Array`, `Map`, `Set`, `Promise` and `Proxy` objects

### CODE REVIEW

File: packages/replay-next/components/inspector/PropertiesRenderer.tsx [hunk 1]
```diff
@@ -52,15 +52,31 @@ export default function PropertiesRenderer({
...
+      }
+      properties.unshift({ name: "<state>", value: preview.promiseState.state });
+    }
+
+    if (preview.proxyState) {
+      properties.unshift(
+        { name: "<target>", ...preview.proxyState.target },
+        { name: "<handler>", ...preview.proxyState.handler }
+      );
+    }
```
Comment (bvaughn) [inline packages/replay-next/components/inspector/PropertiesRenderer.tsx:77]: Can we add a screenshot test for each of these types?

**Review by bvaughn — COMMENTED** [review]
> * fix the display of `Set` contents

I don't see any changes that reference sets directly. Can you elaborate on this please?

**Comment by hbenl** [comment]
> I don't see any changes that reference sets directly.

It's the change shown at the bottom in GitHub's changes view, in `SetContainerEntriesChildrenRenderer`: it used to surround all elements of a `Set` with curly braces.

**Comment by bvaughn** [comment]
>> I don't see any changes that reference sets directly.
>
> It's the change shown at the bottom in GitHub's changes view, in `SetContainerEntriesChildrenRenderer`: it used to surround all elements of a `Set` with curly braces.

Isn't that what Chrome does?

Screen Shot 2023-02-24 at 10 53 01 AM

**Comment by bvaughn** [comment]
Oh! I see what you mean.

Screen Shot 2023-02-24 at 10 57 47 AM

Okay yeah that matches Chrome 😄  Nice!
Screen Shot 2023-02-24 at 10 58 01 AM

**Pull request merged**